### ISSUE TITLE
Make Surface accessible via TAB-navigation

### ISSUE THREAD

**Issue by obuchtala** [substance/substance#488]
While solving issues with focusing of surfaces I found that Surfaces are not accessible via TAB navigation, as one requirement for general accessibility.

**Comment by obuchtala** [comment]
@michael Is there a particular reason that SwitchTextTypeTool has a 'title' prop?
https://github.com/substance/substance/blob/master/packages/prose-editor/ProseEditor.js#L29

To me it looks like that this would not be optional, or should be set be the user/programmer at all.

**Comment by michael** [comment]
I think we can remove that configuration. And simply call this.i18n.t('switch_text') from within SwitchTextTypeTool.

**Comment by michael** [comment]
:+1: for TAB access for surfaces. Only 'problem' that comes to mind is when within a list TAB would be used for indentation and thus conflict with the switch to the next Surface. However whenever we are not in a list context TAB could do the surface switch.

**Comment by obuchtala** [comment]
Cool. Thx

**Comment by obuchtala** [comment]
@michael maybe u want to try out the accessibility branch? should go together well with examples@collab-refactor.
You should be able to cycle through tools and surfaces, and also use tools with SPACE or ENTER.

**Comment by michael** [comment]
Can't do now. but yes will have a look!

**Comment by michael** [comment]
Smooth! Now we enter professional state :) I think we eventually want a unique css class to have a visual difference between active states and focused (via tab). I would draw some kind of border or outline.

Potential issues:
- When you marked undo for instance and activate it with space once you loose the focus of that button
- Elements are not always accessible: E.g. once the address bar is reached via tab you will never find your way back into the menus. Some other time it was randomly not possible. It seems it only works when you start tabbing with a natively focussed Surface.
- Should we only support enter (not space) it seems to be the default in other web pages but not sure I am.

**Comment by obuchtala** [comment]
> - Should we only support enter (not space) it seems to be the default in other web pages but not sure I am

ENTER and SPACE are default obviously. Notice that it is the native behavior, not a custom handler.

**Comment by obuchtala** [comment]
> Elements are not always accessible: E.g. once the address bar is reached via tab you will never find your way back into the menus. Some other time it was randomly not possible. It seems it only works when you start tabbing with a natively focussed Surface.

It should even allow you to cycle through.
However you should never open the console. This seems to 'steal' the navigation somehow.

**Comment by obuchtala** [comment]
> When you marked undo for instance and activate it with space once you loose the focus of that button

Focusing is done natively when rendering the selection, not by us.

I.e. when we apply the change and render the selection, then the surface will have the focus afterwards.

**Comment by obuchtala** [comment]
Some improvements have been done. But accessibility is out-of-scope for now.
